cyberbrain/value_stack.py

```diff
@@ -911,18 +1062,18 @@ def _RAISE_VARARGS_handler(self, instr):
 
     # noinspection DuplicatedCode
     def _WITH_CLEANUP_START_handler(self, exc_info):
-        exc = self.tos
+        exc = self.tos.custom_value
         exit_func: any
-        if not exc:  # Checks if tos is None, which in our stack, is []
+        if exc is None:
             exit_func = self.stack.pop(-2)
         elif isinstance(exc, Why):
             if exc in {Why.RETURN, Why.CONTINUE}:
-                exit_func = self.stack.pop(-2)  # why, ret_val, __exit__
+                exit_func = self.stack.pop(-2).custom_value  # why, ret_val, __exit__
             else:
-                exit_func = self.stack.pop(-1)  # why, __exit__
+                exit_func = self.stack.pop(-1).custom_value  # why, __exit__
         elif utils.is_exception_class(exc):
-            w, v, u = self._pop(3)
-            tp2, exc2, tb2 = self._pop(3)
+            w, v, u = (x.custom_value for x in self._pop(3))
```

Comment:
The quote can be removed

Reply:
This is generator syntax, so parens are needed

Reply:
At least for other similar lines, this line will be reverted to just self._pop

Reply:
You're right, not sure why I left the comment